module/Finna/src/Finna/View/Helper/Root/Citation.php

```diff
@@ -81,6 +104,37 @@ public function getCitationHarvard()
         }
     }
 
+    /**
+     * Get Archive citation.
+     *
+     * This function returns a citation for archive items.
+     *
+     * @return string
+     */
+    public function getCitationArchive(): string
+    {
+        $serverUrl = $this->getView()->plugin('serverUrl');
+        $recordLinker = $this->getView()->plugin('recordLinker');
+        $url = $serverUrl($recordLinker->getUrl($this->driver));
+        $id = $this->driver->tryMethod('getUniqueID');
+        $topId = $this->driver->tryMethod('getHierarchyTopId')[0];
```

Comment:
Miksei haeta suoraan top-titleä (getHierarchyTopTitle)? tryMethod voi palauttaa null:n tai tyhjän taulukon, pitää tarkistaa ennen kuin otetaan siitä ensimmäinen alkio.

Reply:
Tässä oli se vuosiluku asia, kun se ei hae sillä niitä ylimmän tason arkiston vuosia ja kysyin onko parempi vain tehdä getHierarchyTopTitle vai onko ok tehdä näin? (Koska se ei ollut pakollinen lisättävä jos ei itse datassa.)

Reply:
getHierarchyTopTitle palauttaa sen, jolla hierarkian ylätasoon yleensä halutaan viitata. Ei ole välttämättä 1:1 tavallisen nimekkeen kanssa, mutta mitä nyt sitten halutaan?

Reply:
Aa, siis tiketissä olisi että jos alemman tason tietue: arkiston nimi (arkiston vuodet), tietueen nimi (tietueen vuodet)